app/models/smart_proxy.rb

```diff
@@ -82,18 +82,34 @@ def refresh
   end
 
   def taxonomy_foreign_conditions
-    feature_names = self.features.pluck(:name)
-    conditions                      = {}
-    if feature_names.include?('Puppet') && feature_names.include?('Puppet CA')
+    conditions = {}
+    if has_puppet? && has_puppet_ca?
       conditions = "puppet_proxy_id = #{id} OR puppet_ca_proxy_id = #{id}"
-    elsif feature_names.include?('Puppet')
+    elsif has_puppet?
       conditions[:puppet_proxy_id] = id
-    elsif feature_names.include?('Puppet CA')
+    elsif has_puppet_ca?
       conditions[:puppet_ca_proxy_id] = id
     end
     conditions
   end
 
+  # Helper methods for features - e.g. has_puppet? has_dns?
+  def method_missing(method, *args, &block)
```

Comment:
Is there something wrong with this approach? It's used often enough, and it looks nice.  It's railsy.